lib/spack/spack/modules/common.py

```diff
@@ -763,6 +769,36 @@ def has_manpath_modifications(self):
         else:
             return False
 
+    @tengine.context_property
+    def conflicts(self):
+        """List of conflicts for the module file."""
+        fmts = []
+        projection = proj.get_projection(self.conf.projections, self.spec)
+        for item in self.conf.conflicts:
+            self._verify_conflict_naming_consistency_or_raise(item, projection)
+            item = self.spec.format(item)
+            fmts.append(item)
+        return fmts
+
+    def _verify_conflict_naming_consistency_or_raise(self, item, projection):
+        f = string.Formatter()
+        errors = []
+        if len([x for x in f.parse(item)]) > 1:
+            for naming_dir, conflict_dir in zip(projection.split("/"), item.split("/")):
+                if naming_dir != conflict_dir:
+                    errors.extend(
+                        [
+                            f"spec={self.spec.cshort_spec}",
+                            f"conflict_scheme={item}",
+                            f"naming_scheme={projection}",
+                        ]
+                    )
+        if errors:
+            raise ModulesError(
+                message="conflict scheme does not match naming scheme",
+                long_message="\n".join(errors),
```

Comment:
LGTM. I may suggest however a last change to improve the layout of the error message:
```python
                long_message="\n    ".join(errors),
```

Message format will be improved this way, for instance:

```
E           spack.modules.common.ModulesError: conflict scheme does not match naming scheme
E               spec=mpileaks@2.3%gcc@10.2.1~debug~opt+shared+static build_system=generic arch=test-debian6-core2/be7geaa
E               conflict_scheme={name}/{compiler.name}
E               naming_scheme={name}/{version}-{compiler.name}
```

Instead of:
```
E           spack.modules.common.ModulesError: conflict scheme does not match naming scheme
E               spec=mpileaks@2.3%gcc@10.2.1~debug~opt+shared+static build_system=generic arch=test-debian6-core2/be7geaa
E           conflict_scheme={name}/{compiler.name}
E           naming_scheme={name}/{version}-{compiler.name}
```

Reply:
Done, but we need to revisit this in another PR. Since we don't know how and where the error message is printed it would be better that the base error message class takes a list of additional lines without space added where the exception is raised.